src/3rd_party_update.c

```diff
@@ -260,7 +300,43 @@ enum swupd_code third_party_update_main(int argc, char **argv)
 
 	/* update 3rd-party bundles */
 	ret_code = third_party_run_operation_multirepo(cmdline_option_repo, update_repos, SWUPD_NO, "update", steps_in_update);
+	if (ret_code) {
+		goto exit;
+	}
+
+	/* read the current template copy */
+	template_file = third_party_get_binary_path(SWUPD_3RD_PARTY_TEMPLATE_FILE);
```

Comment:
I don't think the template should be in the binary path since that poses name collision issues.

Also given the template is distro content and shouldn't need to be written to (outside of doing a system update) couldn't the file just be in /usr/share/swupd? I think we had talked about this a little before so excuse my memory if you had explained why that kind of thing would be problematic X(.

Reply:
I'm ok with storing it in `/usr/share/swupd` if you think it is more appropriate for it to be there. There is currently no `/usr/share/swupd` directory, so if we store it there it would be the only file there.

Reply:
Ah sorry, /usr/share/defaults/swupd is the folder in use.

Reply:
👌

Reply:
That's true, we shouldn't save it in /opt/3rd-party/bin.
But I think there's a confusion on what this file is. This is not the template that will be used to generate the wrappers. That is still a const in code. This is a file to realize if the template has changed or not. So it shouldn't be installed by clear and we can't install that on /usr/.
I think the best place to put that is in /opt/3rd-party.

Now regarding having the template to be used to create wrappers on /usr, @bryteise do you think it's important to have that? If so, it's independent from this PR.

Reply:
@otaviobp Yea you are right, the file is being used for state at the moment. It isn't content but state data. In that case I'd probably keep it in /var/lib/swupd but I don't remember at the moment if you decided to keep state there or in /opt/3rd-party. That said, I think my confusion stems from seeing it as content and moving it to /usr and taking mtime as the mechanism for deciding when to regenerate the wrappers seems like a reasonable thing to do.